tests/fixtures/optimizer/qualify_tables.sql

```diff
@@ -109,3 +109,8 @@ SELECT * FROM ((SELECT * FROM c.db.t AS t) AS _q_0);
 # title: wrapped subquery without alias joined with a table
 SELECT * FROM ((SELECT * FROM t1) INNER JOIN t2 ON a = b);
 SELECT * FROM ((SELECT * FROM c.db.t1 AS t1) AS _q_0 INNER JOIN c.db.t2 AS t2 ON a = b);
+
+# title: using lateral flatten
+# dialect: snowflake
+SELECT flattened.value FROM z, LATERAL FLATTEN(input => result, outer => true) as flattened;
+SELECT flattened.value FROM c.db.z AS z, LATERAL FLATTEN(input => result, outer => TRUE) AS flattened(FLATTEN(input => result, outer => TRUE) AS SEQ, FLATTEN(input => result, outer => TRUE) AS KEY, FLATTEN(input => result, outer => TRUE) AS PATH, FLATTEN(input => result, outer => TRUE) AS INDEX, FLATTEN(input => result, outer => TRUE) AS VALUE, FLATTEN(input => result, outer => TRUE) AS THIS);
```

Comment:
@edpaget can you explain this test a little? it looks way more complex than expected.

What I had in mind is `... flatten(...) as t(seq, key, ....)`, why do we need to repeat all these `FLATTEN` calls?

Reply:
To answer both this question and the [one here](https://github.com/tobymao/sqlglot/pull/2417#discussion_r1362489989). This allowed the lineage function to find the source for these columns.

Reply:
I agree its bad! Happy to solve this in the parser.

Reply:
What is `lineage` doing when you create a table alias like the one I mentioned above? Why doesn't it work?

Reply:
It stops when it finds the column in the lateral flatten scope and doesn't have a way to find the source lateral flatten is reading from.